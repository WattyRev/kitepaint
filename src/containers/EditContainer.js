import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { getUser } from "../redux/modules/user";
import { getDesignById } from "../redux/modules/designs";
import { getProductById } from "../redux/modules/products";
import { getManufacturerByProductId } from "../redux/modules/manufacturers";
import { GET_DESIGN, GET_PRODUCTS, GET_MANUFACTURERS } from "../redux/actions";
import designShape from "../models/design";
import productShape from "../models/product";
import manufacturerShape from "../models/manufacturer";
import { makeCancelable } from "../utils";

/**
 * Provides information about a specific product.
 */
export class EditContainer extends React.Component {
  static propTypes = {
    /**
     * The ID of the design to get information about.
     */
    designId: PropTypes.string.isRequired,
    /**
     * The design being edited
     */
    design: designShape,
    /**
     * The product. Provided by redux.
     */
    product: productShape,
    /**
     * The manufactuer of the product. Provided by redux.
     */
    manufacturer: manufacturerShape,
    /**
     * The currently logged in user
     */
    user: PropTypes.shape({
      id: PropTypes.string.isRequired
    }),
    /**
     * A function to request the design be fetched. Provided by redux.
     */
    onRequestDesign: PropTypes.func.isRequired,
    /**
     * A function to request the product be fetched. Provided by redux.
     */
    onRequestProduct: PropTypes.func.isRequired,
    /**
     * A function to request the manufacturer of the product be fetched. Provided by redux.
     */
    onRequestManufacturer: PropTypes.func.isRequired,
    /**
     * A function that returns the content to be rendered.
     */
    children: PropTypes.func.isRequired
  };

  state = {
    isLoading: true
  };

  componentDidMount() {
    // Requesting data used on the edit page
    const productRequest = makeCancelable(this.props.onRequestProduct());
    const manufacturerRequest = makeCancelable(
      this.props.onRequestManufacturer()
    );
    const designRequest = makeCancelable(
      this.props.onRequestDesign(this.props.designId)
    );
    this.cancelablePromises.push(productRequest);
    this.cancelablePromises.push(manufacturerRequest);
    this.cancelablePromises.push(designRequest);
    Promise.all([
      productRequest.promise,
      manufacturerRequest.promise,
      designRequest.promise
    ])
      .then(() => {
        this.setState({
          isLoading: false
        });
      })
      .catch(() => {
        this.setState({
          isLoading: false
        });
      });
  }

  componentWillUnmount() {
    this.cancelablePromises.forEach(cancelable => cancelable.cancel());
  }

  cancelablePromises = [];

  render() {
    // If the design was not made by the current user, don't provide it.
    const userMatchesDesign =
      this.props.design && this.props.design.user === this.props.user.id;
    return this.props.children({
      props: {
        design: userMatchesDesign ? this.props.design : null,
        isLoading: this.state.isLoading,
        product: this.props.product,
        manufacturer: this.props.manufacturer,
        user: this.props.user
      }
    });
  }
}

const mapStateToProps = (state, props) => ({
  user: getUser(state),
  product: getProductById(state, props.design && props.design.product),
  manufacturer: getManufacturerByProductId(
    state,
    props.design && props.design.product
  )
});

const mapDispatchToProps = {
  onRequestDesign: GET_DESIGN,
  onRequestProduct: GET_PRODUCTS,
  onRequestManufacturer: GET_MANUFACTURERS
};

const withoutDesign = connect(
  mapStateToProps,
  mapDispatchToProps
)(EditContainer);

export default connect((state, props) => ({
  design: getDesignById(state, props.designId)
}))(withoutDesign);
