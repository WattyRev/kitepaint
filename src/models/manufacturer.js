import PropTypes from "prop-types";

/**
 * A manufacturer is a company that manufactures kites and owns one or more products.
 */
const manufacturerState = PropTypes.shape({
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  logo: PropTypes.string.isRequired,
  website: PropTypes.string
});

export default manufacturerState;

/**
 * A mock manufacturer used for testing
 * @type {Object}
 */
const getMockManufacturer = () => ({
  id: "abc",
  name: "Krazy Kites",
  logo: "abc.jpg",
  website: "http://krazykites.com"
});
export { getMockManufacturer };
