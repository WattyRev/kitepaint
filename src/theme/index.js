import * as colors from "./colors";
import * as patterns from "./patterns";
import A from "./A";
import BlockListItem from "./BlockListItem";
import Button from "./Button";
import Dropdown from "./Dropdown";
import Error from "./Error";
import H1 from "./H1";
import H2 from "./H2";
import H3 from "./H3";
import Icon from "./Icon";
import Input from "./Input";
import Label from "./Label";
import ModalPrompt from "./ModalPrompt";
import P from "./P";
import Text from "./Text";
import Tooltip from "./Tooltip";

const Theme = { colors, patterns };
export default Theme;

export {
  A,
  BlockListItem,
  Button,
  Dropdown,
  Error,
  H1,
  H2,
  H3,
  Icon,
  Input,
  Label,
  ModalPrompt,
  P,
  Text,
  Tooltip
};
