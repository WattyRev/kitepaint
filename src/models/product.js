import PropTypes from "prop-types";
import { getMockManufacturer } from "./manufacturer";
import Status, { statusProp } from "./status";

/**
 * A product is a kite that can be customized to create a design.
 */
const productShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  embed: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  notes: PropTypes.arrayOf(PropTypes.string),
  manufacturer: PropTypes.string.isRequired,
  status: statusProp.isRequired,
  url: PropTypes.string,
  colors: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      color: PropTypes.string.isRequired
    })
  ),
  variations: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      svg: PropTypes.string.isRequired
    })
  )
});

export default productShape;

/**
 * A mock product used for testing
 * @type {Object}
 */
const getMockProduct = () => ({
  id: "product-1",
  embed: "krazykites.com",
  name: "Krazy Kite",
  manufacturer: getMockManufacturer().id,
  url: "http://krazykites.com/krazy-kite",
  colors: [
    {
      name: "red",
      color: "#ff0000"
    },
    {
      name: "black",
      color: "#000000"
    }
  ],
  variations: [
    {
      name: "Standard",
      svg: "<div>Kool Kite</div>"
    }
  ],
  status: Status.PUBLIC
});

export { getMockProduct };
