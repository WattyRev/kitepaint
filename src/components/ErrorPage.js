import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { getAssetUrl } from "../utils";
import { H1, P } from "../theme";

export const StyleWrapper = styled.div`
  text-align: center;
  min-height: calc(100vh - 50px);
  background: url(${getAssetUrl("/errorBackground.jpg")});
  background-size: cover;
  background-repeat: none;
  position: relative;
  padding-top: 10%;
  box-sizing: border-box;
  &:before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: ${props => props.theme.colors.white};
    opacity: 0.6;
  }

  h1 {
    display: block;
    margin: 0 auto;
    width: 100px;
    height: 100px;
    background: ${props => props.theme.colors.black};
    color: ${props => props.theme.colors.silver};
    border-radius: 50%;
    line-height: 100px;
    position: relative;
    margin-bottom: 16px;
  }

  p {
    position: relative;
  }
`;

const ErrorPage = ({ errorCode, errorMessage }) => (
  <StyleWrapper>
    <H1>{errorCode}</H1>
    <P>{errorMessage}</P>
  </StyleWrapper>
);

ErrorPage.defaultProps = {
  errorCode: 404,
  errorMessage: "Page not found"
};

ErrorPage.propTypes = {
  errorCode: PropTypes.number.isRequired,
  errorMessage: PropTypes.string.isRequired
};

export default ErrorPage;
