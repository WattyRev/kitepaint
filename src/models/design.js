import PropTypes from "prop-types";
import { getMockProduct } from "./product";
import Status, { statusProp } from "./status";

/**
 * A design is a user-created entity. It represetns a saved design of a particular product.
 */
const designShape = PropTypes.shape({
  id: PropTypes.string.isRequired,
  // A string representing the created date
  created: PropTypes.string.isRequired,
  // A string representing the updated date
  updated: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  // The ID of the cooresponding product
  product: PropTypes.string.isRequired,
  // The ID of the user that created the design
  user: PropTypes.string,
  variations: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      svg: PropTypes.string.isRequired,
      primary: PropTypes.bool.isRequired
    })
  ),
  // The status of the cooresponding product
  productStatus: statusProp.isRequired,
  // The status of the design
  status: statusProp.isRequired
});

export default designShape;

/**
 * A mock design used for testing
 * @type {Object}
 */
const getMockDesign = () => ({
  id: "design-1",
  created: "01/01/2018",
  updated: "01/02/2018",
  name: "picasso",
  product: getMockProduct().id,
  variations: [
    {
      name: "Standard",
      svg: "<div>picasso standard</div>",
      primary: true
    }
  ],
  status: Status.PRIVATE,
  productStatus: Status.PUBLIC
});
export { getMockDesign };
