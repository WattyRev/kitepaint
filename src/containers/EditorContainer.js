import React from "react";
import PropTypes from "prop-types";
import { fromJS } from "immutable";
import { connect } from "react-redux";
import { CREATE_DESIGN, UPDATE_DESIGN } from "../redux/actions";
import designShape from "../models/design";
import productShape from "../models/product";
import Status from "../models/status";
import { isEmbedded, defaultBackground } from "../constants/embed";
import ErrorPage from "../components/ErrorPage";
import { softCompareStrings, makeCancelable, embedAllowed } from "../utils";

const appliedColorsShape = PropTypes.objectOf(
  PropTypes.shape({
    name: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired
  })
);

const productAppliedColorsShape = PropTypes.objectOf(appliedColorsShape);

export { appliedColorsShape, productAppliedColorsShape };

/**
 * Manages the overall state of the editor. And provides a means to save changes.
 */
export class EditorContainer extends React.Component {
  static propTypes = {
    /**
     * The product being edited
     */
    product: productShape.isRequired,
    /**
     * The default color (name) to be selected. Will default to the first color on the product
     * otherwise.
     */
    defaultColor: PropTypes.string,
    /**
     * An existing design being edited.
     */
    design: designShape,
    /**
     * The default variation (name) to be selected. Will default to the first variation on the
     * product otherwise.
     */
    defaultVariation: PropTypes.string,
    /**
     * A function that renders content
     */
    children: PropTypes.func.isRequired,
    /**
     * A function called when we want to save the design. Provided by Redux.
     */
    onSave: PropTypes.func.isRequired,
    /**
     * A function called when we want to update a design. Provided by Redux.
     */
    onUpdate: PropTypes.func.isRequired
  };

  /**
   * Parses the variations from the provided design in order to determine what colors have been
   * applied.
   * @return {Object}
   */
  _generateAppliedColors() {
    if (!this.props.design) {
      return {};
    }
    const colors = this.props.product.colors;
    return this.props.design.variations.reduce((accumulated, variation) => {
      const { svg, name } = variation;
      const render = new window.DOMParser().parseFromString(svg, "text/xml");
      const panels = render.querySelectorAll("[data-id]");
      const appliedColors = {};
      for (let i = 0; i < panels.length; i++) {
        const panel = panels[i];
        const color = panel.getAttribute("fill");
        if (color) {
          const colorMatch = colors.find(storedColor =>
            softCompareStrings(storedColor.color, color)
          );
          const colorName = colorMatch ? colorMatch.name : color;
          appliedColors[panel.getAttribute("data-id")] = {
            color,
            name: colorName
          };
        }
      }

      accumulated[name] = appliedColors;
      return accumulated;
    }, {});
  }

  constructor(props, ...rest) {
    super(props, ...rest);

    // Use the first color, or the one specified
    let currentColor = props.product.colors[0];
    if (props.defaultColor) {
      currentColor = props.product.colors.find(color =>
        softCompareStrings(color.name, props.defaultColor)
      );
    }

    // Use the first variation or the one specified
    let currentVariation = props.product.variations[0];
    if (props.defaultVariation) {
      currentVariation = props.product.variations.find(variation =>
        softCompareStrings(variation.name, props.defaultVariation)
      );
    }

    this.state = {
      background: defaultBackground || null,
      hideOutlines: false,
      /**
       * The currently selected color.
       * @type {Object}
       */
      currentColor,
      /**
       * The currently selected variation.
       * @type {Object}
       */
      currentVariation,
      /**
       * The currently applied colors.
       * Example
       * {
       *   "my-variation": {
       *     p1: {
       *       name: "black",
       *       color: "#000"
       *     }
       *   }
       * }
       * @type {Object}
       */
      appliedColors: this._generateAppliedColors()
    };
  }

  /*
   * Cancels any pending promises before being unmounted.
   */
  componentWillUnmount() {
    this.cancelablePromises.forEach(promise => promise.cancel());
  }

  /**
   * An array of promises that may need to be cancelled when the component is unmounted
   */
  cancelablePromises = [];

  /**
   * Handles when a different color is selected by updating state.
   * @param  {String} colorName The name of the newly selected color
   */
  handleColorSelection = colorName => {
    const currentColor = this.props.product.colors.find(color =>
      softCompareStrings(color.name, colorName)
    );
    this.setState({
      currentColor
    });
  };

  /**
   * Handles when a different variation is selected by updating state.
   * @param  {String} variationName The name of the newly selected variation
   */
  handleVariationSelection = variationName => {
    const currentVariation = this.props.product.variations.find(variation =>
      softCompareStrings(variation.name, variationName)
    );
    this.setState({
      currentVariation
    });
  };

  /**
   * Set the color for the specified id on the current variation to the current color.
   * @param  {String} id The ID of the panel, taken from data-id on the element.
   */
  handleColorApplied = id => {
    const appliedColorsMap = fromJS(this.state.appliedColors);
    const appliedColors = appliedColorsMap
      .setIn([this.state.currentVariation.name, id], this.state.currentColor)
      .toJS();
    this.setState({
      appliedColors
    });
  };

  /**
   * Generates the design variations based on the product variations and the applied colors.
   * @return {Object[]} Each object contains name, primary, and svg.
   */
  generateDesignVariations = () => {
    const appliedColors = this.state.appliedColors;
    const productVariations = this.props.product.variations;

    // Build each variation
    return productVariations.map((variation, index) => {
      // Render the blank variation from the product in memory
      const render = new window.DOMParser().parseFromString(
        variation.svg,
        "text/xml"
      );
      const colorMap = appliedColors[variation.name] || {};

      // Apply each color to the rendered variation
      Object.keys(colorMap).forEach(id => {
        const color = colorMap[id].color;
        const panel = render.querySelector(`[data-id="${id}"]`);
        panel.setAttribute("fill", color);
      });

      // Get the new SVG string from the render.
      const svg = render.querySelector("svg").outerHTML;

      // Return the variation
      return {
        name: variation.name,
        primary: !index,
        svg
      };
    });
  };

  /**
   * Handles save by parsing data and submitting a request to create a new design. Redirects to that
   * design's edit page when successful.
   * @param  {Object} data must contain name and user(id)
   */
  handleSave = data => {
    const { name, user } = data;
    const design = {
      name,
      user,
      product: this.props.product.id,
      variations: this.generateDesignVariations(),
      status: user === "0" ? Status.PUBLIC : Status.UNLISTED
    };
    const promise = makeCancelable(this.props.onSave(design));
    promise.promise.then(response => {
      const designId = response.data.id;
      if (data.user === "0") {
        // If the design was created anonymously, go to the view page
        window.location.replace(`/view/${designId}`);
      } else {
        window.location.replace(`/edit/${designId}`);
      }
    });
    this.cancelablePromises.push(promise);
  };

  /**
   * Handles update by parsing data and submitting a request to update the design.
   */
  handleUpdate = () => {
    const design = {
      id: this.props.design.id,
      variations: this.generateDesignVariations()
    };
    this.props.onUpdate(design);
  };

  /**
   * Gets the applied colors for the current variation
   */
  getCurrentVariationColors = () => {
    const currentVariationName = this.state.currentVariation.name;
    return this.state.appliedColors[currentVariationName] || {};
  };

  /**
   * Handles auto fill by merging the colors applied to the current variation on top of the colors
   * applied to every other variation.
   */
  handleAutofill = () => {
    const currentColors = this.getCurrentVariationColors();
    const previousAppliedColors = this.state.appliedColors;
    const variations = this.props.product.variations;
    const appliedColors = variations.reduce((accumulated, variation) => {
      accumulated[variation.name] = {
        ...(previousAppliedColors[variation.name] || {}),
        ...currentColors
      };
      return accumulated;
    }, {});
    this.setState({ appliedColors });
  };

  handleChangeBackground = value => this.setState({ background: value });

  handleToggleHideOutlines = () =>
    this.setState({ hideOutlines: !this.state.hideOutlines });

  render() {
    if (
      isEmbedded &&
      this.props.product &&
      !embedAllowed(this.props.product.embed.split(","))
    ) {
      return (
        <ErrorPage
          errorCode={401}
          errorMessage="Embedding of this page is not permitted."
        />
      );
    }
    const data = {
      actions: {
        applyColor: this.handleColorApplied,
        autofill: this.handleAutofill,
        changeBackground: this.handleChangeBackground,
        toggleHideOutlines: this.handleToggleHideOutlines,
        save: this.handleSave,
        selectColor: this.handleColorSelection,
        selectVariation: this.handleVariationSelection,
        update: this.handleUpdate
      },
      props: {
        appliedColors: this.state.appliedColors,
        background: this.state.background,
        hideOutlines: this.state.hideOutlines,
        currentColor: this.state.currentColor,
        currentVariation: this.state.currentVariation,
        currentVariationColors: this.getCurrentVariationColors()
      }
    };
    return this.props.children(data);
  }
}

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  onSave: CREATE_DESIGN,
  onUpdate: UPDATE_DESIGN
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditorContainer);
