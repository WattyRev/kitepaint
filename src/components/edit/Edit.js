import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import EditContainer from "../../containers/EditContainer";
import EditorContainer from "../../containers/EditorContainer";
import { PageLoader } from "../../theme";
import Toolbar from "../editor/Toolbar";
import Sidebar from "../editor/Sidebar";
import Canvas from "../editor/Canvas";
import ProductNotes from "../editor/ProductNotes";
import ErrorPage from "../ErrorPage";

const PageLayout = styled.div`
  display: flex;
  justify-content: stretch;
  position: relative;

  .product-notes {
    position: absolute;
    background: rgba(0, 0, 0, 0.8);
    bottom: 0;
    right: 0;
    padding: 8px;
    max-width: calc(100% - 200px);
  }
`;

/**
 * A coordinating component that builds the CreatNew page.
 * The CreateNew page is provided a productId from the router, and is intended to create a new
 * design based on that product.
 */
const Edit = ({ match }) => (
  <EditContainer designId={match.params.designId}>
    {editData => {
      if (editData.props.isLoading) {
        return <PageLoader />;
      }
      if (!editData.props.design) {
        return <ErrorPage />;
      }
      if (!editData.props.product) {
        return (
          <ErrorPage errorMessage="This product is no longer available." />
        );
      }
      return (
        <EditorContainer
          design={editData.props.design}
          product={editData.props.product}
        >
          {editorData => (
            <React.Fragment>
              <Toolbar
                design={editData.props.design}
                hideOutlines={editorData.props.hideOutlines}
                onAutofill={editorData.actions.autofill}
                onBackgroundChange={editorData.actions.changeBackground}
                onHideOutlines={editorData.actions.toggleHideOutlines}
                onRedo={editorData.actions.redo}
                onReset={editorData.actions.reset}
                onSave={name =>
                  editorData.actions.save({
                    name,
                    user: editData.props.user.id
                  })
                }
                onUndo={editorData.actions.undo}
                onUpdate={editorData.actions.update}
                redoDisabled={!editorData.props.canRedo}
                showSettings
                undoDisabled={!editorData.props.canUndo}
              />
              <PageLayout>
                <Sidebar
                  design={editData.props.design}
                  product={editData.props.product}
                  manufacturer={editData.props.manufacturer}
                  selectedVariation={editorData.props.currentVariation.name}
                  selectedColor={editorData.props.currentColor.name}
                  onColorSelect={editorData.actions.selectColor}
                  onVariationSelect={editorData.actions.selectVariation}
                  appliedColors={editorData.props.appliedColors}
                />
                <Canvas
                  hideOutlines={editorData.props.hideOutlines}
                  background={editorData.props.background}
                  colorMap={editorData.props.currentVariationColors}
                  svg={editorData.props.currentVariation.svg}
                  onClick={editorData.actions.applyColor}
                  currentColor={editorData.props.currentColor.name}
                />
                {editData.props.product.get("notes") &&
                  !!editData.props.product.get("notes").length && (
                    <ProductNotes notes={editData.props.product.get("notes")} />
                  )}
              </PageLayout>
            </React.Fragment>
          )}
        </EditorContainer>
      );
    }}
  </EditContainer>
);

Edit.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      designId: PropTypes.string.isRequired
    })
  })
};

export default Edit;
