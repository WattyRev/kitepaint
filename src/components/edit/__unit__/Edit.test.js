import React from "react";
import { shallow } from "enzyme";
import { getMockDesign } from "../../../models/Design";
import { getMockProduct } from "../../../models/product";
import { getMockManufacturer } from "../../../models/manufacturer";
import EditContainer from "../../../containers/EditContainer";
import EditorContainer from "../../../containers/EditorContainer";
import { PageLoader } from "../../../theme";
import ErrorPage from "../../ErrorPage";
import Edit from "../Edit";

describe("Edit", () => {
  let defaultProps;
  let editData;
  let editorData;
  beforeEach(() => {
    defaultProps = {
      match: {
        params: {
          designId: "abc"
        }
      }
    };
    editData = {
      actions: {
        save: jest.fn(),
        update: jest.fn()
      },
      props: {
        isLoading: false,
        design: getMockDesign(),
        product: getMockProduct(),
        manufacturer: getMockManufacturer()
      }
    };
    editorData = {
      actions: {
        selectColor: jest.fn(),
        selectVariation: jest.fn(),
        changeBackground: jest.fn(),
        toggleHideOutlines: jest.fn()
      },
      props: {
        appliedColors: {},
        currentVariation: {
          name: "Standard",
          svg: ""
        },
        currentColor: {
          name: "Red",
          color: "#FF0000"
        }
      }
    };
  });
  it("renders", () => {
    const wrapper = shallow(<Edit {...defaultProps} />);
    const editContainer = shallow(
      <div>{wrapper.find(EditContainer).prop("children")(editData)}</div>
    );
    shallow(
      <div>
        {editContainer.find(EditorContainer).prop("children")(editorData)}
      </div>
    );
  });
  it("displays a loading state when loading", () => {
    expect.assertions(1);
    editData.props.isLoading = true;
    const wrapper = shallow(<Edit {...defaultProps} />);
    const content = shallow(
      <div>{wrapper.find(EditContainer).prop("children")(editData)}</div>
    );
    expect(content.find(PageLoader)).toHaveLength(1);
  });
  it("displays an error state if no design was provided", () => {
    expect.assertions(1);
    editData.props.design = null;
    const wrapper = shallow(<Edit {...defaultProps} />);
    const content = shallow(
      <div>{wrapper.find(EditContainer).prop("children")(editData)}</div>
    );
    expect(content.find(ErrorPage)).toHaveLength(1);
  });
});
