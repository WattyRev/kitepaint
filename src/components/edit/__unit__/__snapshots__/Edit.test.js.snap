// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit renders 1`] = `
<div>
  Edit
</div>
`;
